tools/delete_imaging_upload.pl

```diff
@@ -988,3 +1032,102 @@ sub deleteUploadsOnFileSystem {
         }
     }
 }
+
+=pod
+
+=head3 getTypesToDelete($dbh, $scanTypeList)
+
+INPUTS:
+                  
+=cut
+sub getScanTypesToDelete {
+	my($dbh, $scanTypeList) = @_;
+	
+	return () if !defined $scanTypeList;
+	
+	my %types = map { $_=> 1 } split(/,/, $scanTypeList);
+	
+    $query = 'SELECT Scan_type, ID FROM mri_scan_type '
+           . 'WHERE Scan_type IN ('
+           . join(',', ('?') x keys %types)
+           . ')';
+    my $validTypesRef = $dbh->selectall_hashref($query, 'Scan_type', undef, keys %types);
+    
+    return map { $_ => defined $validTypesRef->{$_} } keys %types;
+}
+
+sub getTarchiveSeriesIDs {
+	my($dbh, $filesRef) = @_;
+	
+	return [] unless @{ $filesRef->{'files'} };
+	
+	my @fileID = map { $_->{'FileID'} } @{ $filesRef->{'files'} };
+	my $query = 'SELECT tf.TarchiveSeriesID '
+	          . 'FROM tarchive_files tf '
+	          . 'JOIN tarchive_series ts ON (tf.TarchiveSeriesID=ts.TarchiveSeriesID) '
+	          . "JOIN files f ON (ts.SeriesUID=f.SeriesUID AND ABS(f.EchoTime*1000 - ts.EchoTime) < $FLOAT_EQUALS_THRESHOLD) "
+	          . 'WHERE f.FileID IN (' 
+	          . join(',', ('?') x @fileID)
+	          . ')';
+    my $tarchiveFilesRef = $dbh->selectall_arrayref($query, { 'Slice' => {} }, @fileID);
+
+    return [ map { $_->{'TarchiveSeriesID'} } @$tarchiveFilesRef ];
+}
+
+sub deleteTableData {
+	my($dbh, $table, $key, $keyValuesRef, $tmpSQLBackupFile) = @_;
+	
+	return unless @$keyValuesRef;
+	
+	my $query = "DELETE FROM $table WHERE $key IN("
+	          . join(',', ('?') x @$keyValuesRef)
+	          . ')';
+
+    &updateSQLBackupFile($tmpSQLBackupFile, $table, $key, $keyValuesRef) if $tmpSQLBackupFile;
+
+	$dbh->do($query, undef, @$keyValuesRef);	
+}
+
+sub updateSQLBackupFile {
+	my($tmpSQLBackupFile, $table, $key, $keyValuesRef) = @_;
+			
+	# Make sure all keys are quoted before putting them in the Unix command
+    my %quotedKeys;
+	foreach my $k (@$keyValuesRef) {
+	    (my $quotedKey = $k) =~ s/\'/\\'/g;
+        $quotedKeys{$quotedKey} = 1;
+    }
+		
+	# Read the current contents of the backup file
+    open(SQL, "<$tmpSQLBackupFile") or die "Cannot read $tmpSQLBackupFile: $!\n";
+    my @lines = <SQL>;
+    close(SQL);
+		
+	# Run the mysqldump command for the current table and store the
+	# result in $tmpSqlBackupFile (overwrite contents)
+	my $mysqldumpOptions = '--no-create-info --compact --single-transaction --skip-extended-insert';
+	
+	my $warningToIgnore = 'mysqldump: [Warning] Using a password on the command line interface can be insecure.';
+    my $cmd = sprintf(
+        "mysqldump $mysqldumpOptions --where='%s IN (%s)' --result-file=%s  -h %s -p%s -u %s %s %s |& fgrep -v '$warningToIgnore'",
+		$key,
+		join(',', keys %quotedKeys),
+		quotemeta($tmpSQLBackupFile),
+		quotemeta($Settings::db[3]),
+	    quotemeta($Settings::db[2]),
+		quotemeta($Settings::db[1]),
+		quotemeta($Settings::db[0]),
+		$table
+    );
+		
+    system($cmd) != 0
```

Comment:
Tried with a user that do not have mysqldump permission and:
- I get an empty SQL file created
- the script keep going without stopping and delete everything

Reply:
In the end, I don't have a user without mysqldump permission anywhere so I could not really test that the script indeed dies if the mysqldump command failed...